### PR TITLE
Enable Transactions page SSR

### PR DESCRIPTION
**Pull request by kewitz**
Related to https://github.com/opencollective/opencollective/issues/3385

### CODE REVIEW

File: pages/transactions.js [hunk 1]
```diff
@@ -106,4 +213,15 @@ class TransactionsPage extends React.Component {
   }
 }
 
-export default withUser(addCollectiveCoverData(TransactionsPage));
+const addTransactionsData = graphql(transactionsQuery, {
+  skip: props => !props.slug,
+  name: 'transactionsQuery',
```
Comment (znarf) [inline pages/transactions.js:218]: This is not a good name, it's confusing.

Comment (Betree) [reply pages/transactions.js:218]: I agree that it creates confusion with the query itself, something like `transactionsQueryResult` would make more sense to me.

File: pages/transactions.js [hunk 2]
```diff
@@ -106,4 +213,15 @@ class TransactionsPage extends React.Component {
   }
 }
 
-export default withUser(addCollectiveCoverData(TransactionsPage));
+const addTransactionsData = graphql(transactionsQuery, {
+  skip: props => !props.slug,
```
Comment (Betree) [inline pages/transactions.js:217]: `collectiveSlug` is required in https://github.com/opencollective/opencollective-frontend/blob/a34c29ae3056d356bf760664220b4efedbce6821/server/pages.js#L41. Is this check still necessary?

File: pages/transactions.js (hunk 1 above)
Comment (kewitz) [reply pages/transactions.js:218]: I renamed it to `transactionData`.

Comment (znarf) [reply pages/transactions.js]: Looks better!

**Pull request merged**